### ISSUE TITLE
Allow report of missing fields

### ISSUE THREAD

**Issue by domna** [FAIRmat-NFDI/pynxtools#167]
When the reader runs and does not find required fields it should not stop, but rather set the @partial attribute and report which fields are missing. There could also be a feature which just reports which fields are missing without writing a file. See #133

**Comment by RubelMozumder** [comment]
PR: https://github.com/FAIRmat-NFDI/pynxtools/pull/175

**Comment by domna** [comment]
> PR: #175

Its better to write `Fixes #175` in the PR, because then github automatically links the PR and closes the issue automatically as soon as the branch is merged to the default branch.

**Comment by RubelMozumder** [comment]
@sherjeelshabih, Yeah, we can close it, I found the logger is working perfectly both in Nomad and pynxtools when running converter from jupyter lab or the command line. REader user and developer will be aware of any missing fields, groups or attributes.

**Comment by domna** [comment]
@sherjeelshabih I'm not sure whether we should keep it open as we don't error out with the reader anymore but we don't implemented the part of writing the `@partial` attribute. I think it would be useful to denote in the produced file that this is not a valid NeXus file at least.

**Comment by RubelMozumder** [comment]
> @sherjeelshabih I'm not sure whether we should keep it open as we don't error out with the reader anymore but we don't implemented the part of writing the `@partial` attribute. I think it would be useful to denote in the produced file that this is not a valid NeXus file at least.

Sorry, I did not notice it. Just a short comment on the partial attributes. Alongside with the `@partial`, you might also think to have the list of the missing fields on the parent data object let say under `@missing_concepts`. So later on without the log info, viewer or user might also get deeper info for `@partial` from the `nexus` file in any platform.

**Comment by sherjeelshabih** [comment]
Feel free to reopen this issue as you see fit. I guess we have added functionality to pick up files with the @partial list and merge them. But this as you say is about writing out the @partial list automatically.

**Comment by lukaspie** [comment]
Closed as `@partial` is currently not accepted, we should open a new issue if/when we try to restart this idea.
